### ISSUE TITLE
Create a Scaffolder for ASP.NET API controllers

### ISSUE THREAD

**Issue by matthewrdev** [mfractor/mfractor-feedback#242]
## Criteria

 * Project should reference ASP.NET assemblies.
 * File name should end with `Controller`.

## Template

Inputs:

 * `$namespace$`: The project default namespace and folder path.
 * `$name$`: The name of the new controller.
 * `$route$`: The `$name$` with the `Controller` suffix removed.

```
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.AspNetCore.Mvc;

namespace $namespace$
{
    [ApiController, Route("api")]
    public class $name$ : Controller
    {
        [HttpGet, Route("$route$")]
        public IActionResult Get()
        {
            // TODO: Implement 
            return Ok("Success");
        }
    }
}

**Comment by matthewrdev** [comment]
@ravero Can you comment on this and let me know if this scaffolder and it's template makes sense for everyday use?

**Comment by ravero** [comment]
A few things I would adjust:

* The `HttpGet` attribute already receives route data through the first argument (which is optional), so the `Route` attribute is unnecessary. You can just write like:

`[HttpGet("$route$")]`

* I would split attributes on separate lines (ApiController, Route), its more a personal preference but I think its more legible
* Normally I would inherit from `ControllerBase` on API Controllers
* Since ASP.NET Core 2.2 Microsoft recommends using `ActionResult<T>` instead of `IActionResult`. The `T` can be of the type you will be returning and this will works better with the documentation generation libraries like Swashbuckle (there are a few examples of that in the CRM).

**Suggestion**: as I understand the scaffolder as a tool for quickly creating useful files, a would add a few additional templates:

* An async Action method template, where the method declaration would be:

```
using System.Threading.Tasks;

[HttpGet, Route("$route$")]
public async Task<ActionResult> GetAsync()
{
            // TODO: Implement 
            return Ok("Success");
}
```

* A sync/async alternates of the template with `Post` method instead of Get.

**Comment by matthewrdev** [comment]
This has been completed and will be available in MFrator v4.2.2
